--- lib/modernish/mdl/_IN/sig.mm	2020-01-16 02:02:33.000000000 +0000
+++ lib/modernish/mdl/_IN/sig.mm	2020-02-09 01:42:44.000000000 +0000
@@ -42,12 +42,8 @@
 		_Msh_arg2sig_sanitise || return 1
 		case ${_Msh_sig} in
 		( DIE ) use -q var/stack/trap || return 1
-			if isset -i && ! insubshell; then	# on an interactive shell,
-				_Msh_sig=INT			# ... alias DIE to INT.
-			else
-				_Msh_sigv=DIE
-				return
-			fi ;;
+			_Msh_sigv=DIE
+			return ;;
 		( EXIT )_Msh_sigv=EXIT
 			return ;;
 		( ERR )	if thisshellhas TRAPZERR; then
