--- lib/modernish/cap/BUG_APPENDC.t	2020-02-09 02:09:50.000000000 +0000
+++ lib/modernish/cap/BUG_APPENDC.t	2020-02-09 02:10:28.000000000 +0000
@@ -32,10 +32,7 @@
 		{ : >> "$D/file"; } 2>/dev/null
 	)
 	_Msh_test=$?
-	case $- in
-	( *i* )	PATH=$DEFPATH command rm -rf "${_Msh_testD}" ;;
-	( * )	PATH=$DEFPATH command rm -rf "${_Msh_testD}" & ;;
-	esac
+	PATH=$DEFPATH command rm -rf "${_Msh_testD}" &
 	unset -v _Msh_testD
 	case ${_Msh_test} in
 	( 0 )	return 1 ;;
