--- lib/modernish/cap/QRK_ANDORBG.t	2020-01-07 15:24:07.000000000 +0000
+++ lib/modernish/cap/QRK_ANDORBG.t	2020-02-09 01:42:44.000000000 +0000
@@ -16,10 +16,7 @@
 #	http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_09_03
 #	zsh-workers 44705: http://www.zsh.org/mla/workers/2019/msg00741.html
 
-case $- in
-( *m* )	{ _Msh_test=$(_Msh_test=QRK && : & putln "${_Msh_test-}"); } 2>/dev/null ;;
-( * )	_Msh_test=QRK && : & ;;
-esac
+_Msh_test=QRK && : &
 
 case ${_Msh_test-} in
 ( QRK )	;;
